### PR TITLE
DMA: support for complete DMA sequence

### PR DESCRIPTION
**Pull request by pthierry-ledger**
- [x] support for sys_dma_stream_start()
- [x] support for sys_dma_stream_suspend()
- [x] support for sys_dma_stream_resume()
- [x] full waitforevent support of DMA events
- [x] support DMA manager state automaton upgrade on interrupts events

NOTE:
-  sys_dma_get_status() is out of this PR
- enhanced autotest checks of suspend/resume are out of this PR

### CODE REVIEW

File: autotest/src/tests/test_signal.c [hunk 1]
```diff
@@ -13,7 +13,7 @@ void test_signal_sendrecv(void)
 {
     Status ret;
     taskh_t handle = 0;
-    uint32_t timeout = 20UL;
+    int32_t timeout = 20UL;
```
Comment (fvalette-ledger) [inline autotest/src/tests/test_signal.c:16]: Thus, use only `L` suffix

File: kernel/src/managers/dma/dma.c [hunk 2]
```diff
@@ -384,8 +435,68 @@ kstatus_t mgr_dma_stream_start(const dmah_t dmah)
...
+
+    if (unlikely(cfg == NULL)) {
+        goto end;
+    }
+    /* config entry, when found, must have its meta field properly set (dma_init time set) */
+    /*@ assert \valid_read(cfg); */
+    /*@ assert \valid_read(cfg->meta); */
+
+    /* can't unassign a stream that is started or already unassigned */
+    if (unlikely(
```
Comment (fvalette-ledger) [inline kernel/src/managers/dma/dma.c:451]: weird format

File: kernel/src/managers/dma/dma.c [hunk 3]
```diff
@@ -384,8 +435,68 @@ kstatus_t mgr_dma_stream_start(const dmah_t dmah)
...
+          (cfg->state != DMA_STREAM_STATE_SUSPENDED)
+        )) {
```
Comment (fvalette-ledger) [inline kernel/src/managers/dma/dma.c:491]: itou

File: kernel/src/managers/dma/dma.c (hunk 2 above)
Comment (pthierry-ledger) [reply kernel/src/managers/dma/dma.c:451]: right... old check not cleanly removed. fixed.

File: kernel/src/managers/dma/dma.c (hunk 3 above)
Comment (pthierry-ledger) [reply kernel/src/managers/dma/dma.c:491]: fixed.

**Review by fvalette-ledger — APPROVED** [review]

**Pull request merged**